qa/workunits/rgw/test_rgw_reshard.py

```diff
@@ -0,0 +1,234 @@
+#!/usr/bin/python3
+
+import logging as log
+import time
+import subprocess
+import json
+
+"""
+Rgw manual and dynamic resharding  testing against a running instance
+"""
+# The test cases in this file have been annotated for inventory.
+# To extract the inventory (in csv format) use the command:
+#
+#   grep '^ *# TESTCASE' | sed 's/^ *# TESTCASE //'
+#
+#
+
+log.basicConfig(level=log.DEBUG)
+log.getLogger('botocore').setLevel(log.CRITICAL)
+log.getLogger('boto3').setLevel(log.CRITICAL)
+log.getLogger('urllib3').setLevel(log.CRITICAL)
+
+
+""" Constants """
+USER = 'tester'
+DISPLAY_NAME = 'Testing'
+ACCESS_KEY = 'NX5QOQKC6BH2IDN8HC7A'
+SECRET_KEY = 'LnEsqNNqZIpkzauboDcLXLcYaWwLQ3Kop0zAnKIn'
+BUCKET_NAME1 = 'myfoo'
+BUCKET_NAME2 = 'mybar'
+VER_BUCKET_NAME = 'myver'
+ENDPOINT = 'http://localhost:80'
+
+
+def exec_cmd(cmd):
+    try:
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+        out, err = proc.communicate()
+        if proc.returncode == 0:
+            log.info('command succeeded')
+            if out is not None: log.info(out)
+            return out
+        else:
+            raise Exception("error: %s \nreturncode: %s" % (err, proc.returncode))
+    except Exception as e:
+        log.error('command failed')
+        log.error(e)
+        return False
+
+
+def install_reqs():
+    out = exec_cmd('cat /etc/*release')
+    result = {}
+    for row in out.decode('utf8').split('\n'):
+        if '=' in row:
+            k, v = row.split('=')
+            result[k.strip()] = v.strip('""')
+    if result['NAME'] == 'Ubuntu':
+        exec_cmd('sudo apt install -y python3-pip')
+    else:
+        exec_cmd('sudo yum install -y python3-pip')
+    exec_cmd('sudo pip3 install boto3')
```

Comment:
this shouldn't be installing stuff with sudo. if we need to set up some dependencies on the teuthology worker, i think we're better off creating a wrapper shell script that sets up a virtualenv and runs this script in it

Reply:
Fixed. 
http://pulpito.ceph.com/smanjara-2020-04-30_09:06:32-rgw:verify-wip-dynamic-resharding-distro-basic-smithi/